pennylane/transforms/hamiltonian_expand.py

```diff
@@ -170,7 +170,7 @@ def hamiltonian_expand(tape, group=True):
     def processing_fn(res):
         # note: res could have an extra dimension here if a shots_distribution
         # is used for evaluation
-        dot_products = [qml.math.dot(qml.math.squeeze(r), c) for c, r in zip(coeffs, res)]
+        dot_products = [qml.math.dot(qml.math.squeeze(res[i]), coeffs[i]) for i in range(len(res))]
```

Comment:
I noticed a bug here

Reply:
what was the bug?

Reply:
That was the whole tensorflow problem for which I opened another PR in the meantime!